lib/AST/ASTContext.cpp

```diff
@@ -1291,6 +1291,12 @@ ASTContext::getModule(ArrayRef<std::pair<Identifier, SourceLoc>> ModulePath) {
     return M;
 
   auto moduleID = ModulePath[0];
+  // Record this load request.  If we've seen it before and couldn't produce a
+  // loaded module it must have failed to load before.
+  if (!RequestedModuleLoads.insert(moduleID.first).second) {
```

Comment:
```c++
llvm::MapVector<Identifier, ModuleDecl*> LoadedModules;
```

If `moduleID.first` is in the map, but the value is `nullptr`, then it must have failed to load. (This would require changing some of the other places that use LoadedModules too.)